src/integrationTest/java/org/radarcns/sink/hdfs/IntegrationTest.java

```diff
@@ -0,0 +1,94 @@
+import org.radarcns.producer.rest.SchemaRetriever;
+import org.radarcns.topic.AvroTopic;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IntegrationTest {
+    @Test(timeout = 120_000L)
+    public void integrationTest() throws IOException, InterruptedException {
+        RestSender sender = new RestSender.Builder()
+                .server(new ServerConfig("http://localhost:8082"))
+                .schemaRetriever(new SchemaRetriever(
+                        new ServerConfig("http://localhost:8081"), 5))
+                .build();
+
+        AvroTopic<ObservationKey, PhoneLight> test1 = new AvroTopic<>("test",
+                ObservationKey.getClassSchema(), PhoneLight.getClassSchema(),
+                ObservationKey.class, PhoneLight.class);
+
+        Schema updatePhoneSchema = new Schema.Parser().parse("{\"type\":\"record\",\"name\":\"PhoneLight\",\"namespace\":\"org.radarcns.passive.phone\",\"doc\":\"Data from the light sensor in luminous flux per unit area.\",\"fields\":[{\"name\":\"time\",\"type\":\"double\",\"doc\":\"Device timestamp in UTC (s).\"},{\"name\":\"timeReceived\",\"type\":\"double\",\"doc\":\"Device receiver timestamp in UTC (s).\"},{\"name\":\"light\",\"type\":[\"null\",\"float\"],\"doc\":\"Illuminance (lx).\"}]}");
+
+        AvroTopic<ObservationKey, PhoneLight> test2 = new AvroTopic<>("test",
+                ObservationKey.getClassSchema(), updatePhoneSchema,
+                ObservationKey.class, PhoneLight.class);
+
+        AvroTopic<ObservationKey, PhoneSmsUnread> test3 = new AvroTopic<>("test",
+                ObservationKey.getClassSchema(), PhoneSmsUnread.getClassSchema(),
+                ObservationKey.class, PhoneSmsUnread.class);
+
+        try (KafkaTopicSender<ObservationKey, PhoneLight> topicSender = sender.sender(test1)) {
+            topicSender.send(new ObservationKey("a", "b", "c"), new PhoneLight(1d, 1d, 1f));
+        }
+        try (KafkaTopicSender<ObservationKey, PhoneLight> topicSender = sender.sender(test2)) {
+            topicSender.send(new ObservationKey("a", "b", "c"), new PhoneLight(1d, 1d, 1f));
+        }
+
+        try (KafkaTopicSender<ObservationKey, PhoneSmsUnread> topicSender = sender.sender(test3)) {
+            topicSender.send(new ObservationKey("a", "b", "c"), new PhoneSmsUnread(1d, 1d, 1));
+        }
+
+        Path path = new Path("hdfs://localhost/topicAndroidNew");
+
+        Configuration conf = new Configuration();
+        do {
+            Thread.sleep(1000);
+
+            List<String> filePaths = getAllFilePath(path, path.getFileSystem(conf))
+                    .stream()
+                    .filter(s -> !s.contains("+tmp"))
+                    .collect(Collectors.toList());
+            if (filePaths.size() >= 3) {
+                System.out.println(filePaths);
+                break;
+            }
+        } while (true);
```

Comment:
Can we have a max retries constraint here to prevent infinite loop because of any problems

Reply:
There is a timeout on the method limiting it to two minutes.

Reply:
Ah okay, missed that. Thanks